### PR TITLE
feat(websocket): remove deflate compression option

### PR DESCRIPTION
**Pull request by thomaseizinger**
## Description

This option is the only reason we pull in a dependency on the shared `zlib` library. Unfortunately, we cannot use a pure Rust backend here because that one does not support configuring the window bits of the deflate algorithm which is used in the deflate websocket extension.

Using websockets in libp2p is already crazy inefficient because you end up with double encryption when using `/wss` (which is enforced by browsers if your website is served via https). A good encryption algorithm like noise or TLS produces an output that looks completely random. Any attempt in compressing this is pretty much wasted CPU power.

Thus, I think removing this configuration option does not really hurt and allows us to remove the dependency on the `zlib` shared library.

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Review by mxinden — APPROVED** [review]
Good thinking.

Missing a changelog entry.

File: transports/websocket/CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,12 @@
 - Raise MSRV to 1.65.
   See [PR 3715].
 
+- Remove `WsConfig::use_deflate` option.
+  This allows us to remove the dependency on the `zlib` shared library.
+  See [PR XXXX].
+
 [PR 3715]: https://github.com/libp2p/rust-libp2p/pull/3715
+[PR XXXX]: https://github.com/libp2p/rust-libp2p/pull/XXXX
```
Comment (thomaseizinger) [inline transports/websocket/CHANGELOG.md]:
```suggestion
  See [PR 3949].

[PR 3715]: https://github.com/libp2p/rust-libp2p/pull/3715
[PR 3949]: https://github.com/libp2p/rust-libp2p/pull/3949
```

@mxinden There is an entry here, would you like another one somewhere else?

**Comment by folex** [comment]
What's the alternative way to enable compression for a protocol? Or should it be done on a per-protocol level manually?

**Comment by thomaseizinger** [comment]
> What's the alternative way to enable compression for a protocol? Or should it be done on a per-protocol level manually?

You are better off compressing the data before writing it to the stream or generally use a more space-efficient encoding / data format.

Compressing encrypted data makes little sense so you can't apply it generically.

**Pull request merged**